### ISSUE TITLE
Arbitrary Ranking Metric

### ISSUE THREAD

**Issue by R2D2VaderBeef** [sectorsedgecommunity/PUG-Bot#2]
At SETO, we'd developed a couple of score formulas, not designed for PUGs specifically:

Formula type 1:

T = Tournament Type
S = Tournament Size
P = Player Position
M = No of matches they played
W = No of wins they had

Score (added on at the end of a tournament) = (S-P) * (W/M) * T

Formula type 2:

T = Tournament Type
O = Average score of Opponent's team
P = Player's Score
F = Leaderboard Placement in match
L = Amount of players in match

Score gained on match win = (O/P) * (L/F) * T
Score lost on match loss = (P/O) * (F/L) * T

This requires further testing on the maths to see if either one is suitable for round robin PUGs. The "tournament type" factor can be ignored for this use case alone, however if we want to ever merge the data back to SETO, it may be useful (although maybe it can be defined as 1 for PUGs and a higher number for other tournament types).

**Comment by R2D2VaderBeef** [comment]
Formula type 1 has a couple of drawbacks - unsurprising, as it was created for elimination modes. I initially tested it multiplying the formula output by 10, rounded to the nearest 1, to make the numbers easier to digest.

- The higher your position (rewarded by S-P), the higher your match win percentage (W/M) will be, so the points are heavily weighted in favour of the winners. If you win, you win big, and everyone else gets very little compared to that.

- In some scenarios, you get less points for winning the same number of games in a larger tournament, even if those games were the exact same difficulty.

I've come up with an improved formula which partially fixes both of these: `(S-P) * (W/M + 1) * 5 (rounded to the nearest 1)`.

**Comment by R2D2VaderBeef** [comment]
Using `Score = (S-P) * (W/M + K) * 10 / (K+1) (rounded to the nearest 1)` where K = 0 for single elim and K = 1 for round robin is another possibility.
